File: frontend/src/api/cartApi.ts
import { CartResponse } from '../types/Cart';
import axiosClient from './axiosClient.ts';

export const cartApi = {
    getCart: async (): Promise<CartResponse> => {
        return await axiosClient.get('/cart');
    },

    updateQuantity: async (productId: string, quantity: number): Promise<CartResponse> => {
        return await axiosClient.put('/cart/update', { productId, quantity });
    },

    removeItem: async (productId: string): Promise<CartResponse> => {
        return await axiosClient.delete(`/cart/remove/${productId}`);
    },

    clearCart: async (): Promise<CartResponse> => {
        return await axiosClient.delete('/cart/clear');
    },

    addToCart: async (productId: string, quantity: number = 1): Promise<CartResponse> => {
        return await axiosClient.post('/cart/add', { productId, quantity });
    }
}

File: Mobileptit/app/src/main/java/com/uteshop/app/data/remote/api/CartApi.java
package com.uteshop.app.data.remote.api;

import com.uteshop.app.data.remote.dto.request.*;
import com.uteshop.app.data.remote.dto.response.*;
import retrofit2.Call;
import retrofit2.http.*;

public interface CartApi {
    @GET("v1/api/cart")
    Call<CartResponse> getCart();

    @GET("v1/api/cart/count")
    Call<CartCountResponse> getCartCount();

    @POST("v1/api/cart/add")
    Call<CartMutationResponse> addToCart(@Body AddToCartRequest r);

    @PUT("v1/api/cart/update")
    Call<CartUpdateResponse> updateCartItem(@Body UpdateCartItemRequest r);

    @DELETE("v1/api/cart/remove/{productId}")
    Call<CartMutationResponse> removeFromCart(@Path("productId") String productId);

    @DELETE("v1/api/cart/clear")
    Call<SimpleResponse> clearCart();
}
